File: Diagram/TransitionDiagramClipData.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Imports System.Drawing

<Serializable()>
Class TransitionDiagramClipData
    Public Entries As New List(Of TransitionDiagramClipDataEntry)
End Class

<Serializable()>
Class DeterministicTransitionClipData

    Public StratumSource As String
    Public StateClassSource As String
    Public StratumDest As String
    Public StateClassDest As String

    Public StratumIdSource As Nullable(Of Integer)
    Public StateClassIdSource As Integer
    Public StratumIdDest As Nullable(Of Integer)
    Public StateClassIdDest As Nullable(Of Integer)
    Public AgeMin As Nullable(Of Integer)
    Public AgeMax As Nullable(Of Integer)

End Class

<Serializable()>
Class ProbabilisticTransitionClipData

    Public StratumSource As String
    Public StateClassSource As String
    Public StratumDest As String
    Public StateClassDest As String
    Public TransitionType As String

    Public StratumIdSource As Nullable(Of Integer)
    Public StateClassIdSource As Integer
    Public StratumIdDest As Nullable(Of Integer)
    Public StateClassIdDest As Nullable(Of Integer)
    Public TransitionTypeId As Integer
    Public Probability As Double
    Public Proportion As Nullable(Of Double)
    Public AgeMin As Nullable(Of Integer)
    Public AgeMax As Nullable(Of Integer)
    Public AgeRelative As Nullable(Of Integer)
    Public AgeReset As Nullable(Of Boolean)
    Public TstMin As Nullable(Of Integer)
    Public TstMax As Nullable(Of Integer)
    Public TstRelative As Nullable(Of Integer)

End Class

<Serializable()>
Class TransitionDiagramClipDataEntry

    Public ShapeData As New DeterministicTransitionClipData
    Public IncomingDT As New List(Of DeterministicTransitionClipData)
    Public IncomingPT As New List(Of ProbabilisticTransitionClipData)
    Public OutgoingPT As New List(Of ProbabilisticTransitionClipData)
    Public Row As Integer
    Public Column As Integer
    Public Bounds As Rectangle

End Class

File: src/Diagram/TransitionDiagramClipData.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2019 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Drawing;
using System.Collections.Generic;

namespace SyncroSim.STSim
{
    [Serializable()]
    internal class TransitionDiagramClipData
    {
        public List<TransitionDiagramClipDataEntry> Entries = new List<TransitionDiagramClipDataEntry>();
    }

    [Serializable()]
    internal class DeterministicTransitionClipData
    {
        public string StratumSource;
        public string StateClassSource;
        public string StratumDest;
        public string StateClassDest;

        public int? StratumIdSource;
        public int StateClassIdSource;
        public int? StratumIdDest;
        public int? StateClassIdDest;
        public int? AgeMin;
        public int? AgeMax;
    }

    [Serializable()]
    internal class ProbabilisticTransitionClipData
    {
        public string StratumSource;
        public string StateClassSource;
        public string StratumDest;
        public string StateClassDest;
        public string TransitionType;

        public int? StratumIdSource;
        public int StateClassIdSource;
        public int? StratumIdDest;
        public int? StateClassIdDest;
        public int TransitionTypeId;
        public double Probability;
        public double? Proportion;
        public int? AgeMin;
        public int? AgeMax;
        public int? AgeRelative;
        public bool? AgeReset;
        public int? TstMin;
        public int? TstMax;
        public int? TstRelative;
    }

    [Serializable()]
    internal class TransitionDiagramClipDataEntry
    {
        public DeterministicTransitionClipData ShapeData = new DeterministicTransitionClipData();
        public List<DeterministicTransitionClipData> IncomingDT = new List<DeterministicTransitionClipData>();
        public List<ProbabilisticTransitionClipData> IncomingPT = new List<ProbabilisticTransitionClipData>();
        public List<ProbabilisticTransitionClipData> OutgoingPT = new List<ProbabilisticTransitionClipData>();
        public int Row;
        public int Column;
        public Rectangle Bounds;
    }
}
